### PR TITLE
GH1533: Updated DotNetCoreBuild inline with Tooling v1.0

### PR DESCRIPTION
**Pull request by JonCubed**
* Removed settings that are no longer valid, if you prefer for them to be marked as obsolete instead let me know.
* Updated unit tests

See #1533 for more info

### CODE REVIEW

File: src/Cake.Common/Tools/DotNetCore/Build/DotNetCoreBuildSettings.cs [hunk 1]
```diff
@@ -42,11 +37,6 @@ public sealed class DotNetCoreBuildSettings : DotNetCoreSettings
         public string VersionSuffix { get; set; }
```
Comment (devlead) [inline src/Cake.Common/Tools/DotNetCore/Build/DotNetCoreBuildSettings.cs]: We should add a string Version { get; set; } too, equivalent to
```csharp
 ArgumentCustomization = arg=>arg.Append(string.Concat("/p:Version=", semVersion)),
``` 

We should also add an 
```csharp
public IDictionary<string, IList<string>> Properties => _properties;
```
To support MSBuild properties without ArgumentCustomization, similar too [MSBuildSettings.Properties](https://github.com/cake-build/cake/blob/1eb28cf18637a1316c8f10327b9d70b9975507ea/src/Cake.Common/Tools/MSBuild/MSBuildSettings.cs#L32)

We should also add an 
```csharp
public ISet<string> Targets { get; }
```

Running `dotnet msbuild -h` looks like it support most MSBuild so future refactoring could be to see what we have common between the .NET Core CLI and MSBuild alias.

Comment (devlead) [inline src/Cake.Common/Tools/DotNetCore/Build/DotNetCoreBuildSettings.cs]: Maybe version could be something common, know `dotnet pack` accepts & should have it too (haven't all yet but probably publish too).

Comment (patriksvensson) [inline src/Cake.Common/Tools/DotNetCore/Build/DotNetCoreBuildSettings.cs]: @devlead Should we really add `Version` now? Isn't it better to wait until the API settled a little bit? I would rather see MSBuild properties first.

If we add version we should add support for `AssemblyVersion` and `FileVersion` as well.

Comment (JonCubed) [inline src/Cake.Common/Tools/DotNetCore/Build/DotNetCoreBuildSettings.cs]: I was planning to add `VersionPrefix` after all the current tasks are done, along with `Properties`. I don't think we should use `Version`, I think it gets overridden in some situations unexpectedly and doesn't work with `VersionSuffix`.  

@patriksvensson if my memory is correct, when I was testing using `VersionPrefix` with the build command acted as `AssemblyVersion`

**Comment by devlead** [comment]
@JonCubed  your changes have been merged, thanks for your contribution 👍

**Pull request merged**